spec/index.bs

```diff
@@ -6767,19 +6787,40 @@ interface GPUCanvasContext {
         given device.
 
         <div algorithm="GPUCanvasContext.getSwapChainPreferredFormat">
-            **Called on:** {{GPUCanvasContext}} this.
+            **Called on:** {{GPUCanvasContext}} |this|.
 
             **Arguments:**
-            <pre class=argumentdef for="GPUCanvasContext/getSwapChainPreferredFormat(device)">
-                device: Device the swap chain format should be queried for.
+            <pre class=argumentdef for="GPUCanvasContext/getSwapChainPreferredFormat(adapter)">
+                |adapter|: Adapter the swap chain format should be queried for.
             </pre>
 
-            **Returns:** {{Promise}}&lt;{{GPUTextureFormat}}&gt;
+            **Returns:** {{GPUTextureFormat}}
 
-            Issue: Describe {{GPUCanvasContext/getSwapChainPreferredFormat()}} algorithm steps.
+            <div class=content-timeline>
+                1. Let |preferredFormat| be the [$preferred swap chain format$] for |this| and |adapter|.
+                1. Return |preferredFormat|.
+            </div>
         </div>
 </dl>
 
+The <dfn dfn>supported swap chain formats</dfn> are a [=set=] of {{GPUTextureFormat}}s that must be
+supported when specified as a {{GPUSwapChainDescriptor}}.{{GPUSwapChainDescriptor/format}} regardless
+of the given {{GPUSwapChainDescriptor}}.{{GPUSwapChainDescriptor/device}}, initially set to:
+&laquo;{{GPUTextureFormat/"bgra8unorm"}}, {{GPUTextureFormat/"bgra8unorm-srgb"}},
+{{GPUTextureFormat/"rgba8unorm"}}, {{GPUTextureFormat/"rgba8unorm-srgb"}}&raquo;.
+
+<div algorithm>
+    To get the <dfn abstract-op>preferred swap chain format</dfn> for a given {{GPUCanvasContext}}
+    |context| and a given {{GPUAdapter}} |adapter| run the following steps::
+
+    1. Let |preferredFormat| be |context|.{{GPUCanvasContext/[[preferred_format]]}}
+    1. If |preferredFormat|[|adapter|] does not exist:
```

Comment:
Just a thought, now that [[preferred_format]] is only used here, all it does is guarantee that the browser will always return the same value for every (canvas context, adapter) pair. I'm sort of suspecting in the future it would look something more like:

- Adapter has three internal slots, [[preferred_default_format]], [[preferred_srgb_format]], and [[preferred_hdr_format]], and getSwapChainPreferredFormat would return one of those three depending on the properties of the canvas context.

but of course we don't want to encode all that right now. But maybe with some confirmation (from e.g. @jdashg), we could put the preferred format directly on the adapter for now? (i.e. say that the preferred format doesn't depend on the canvas, and not worry about that until we have srgb and hdr canvases)